docs/JavaCodePerformance.md

```diff
@@ -551,6 +551,37 @@ class Good {
 ```
 **See:** [IUOXAR09: XML related `XXXFactory.newInstance()` is called repeatedly.](JavaCodePerformance.md#IUOXAR09)   
 
+#### IBI24
+**Observation: HttpRoute is constructed without specifying the secure argument.**   
+**Problem:** For Apache HttpRoute, if you don't specify whether the route is secure, the default of non-secure is taken. This is unclear. If you use it to configure a connection manager for that route, and the actual route is secure,
+the key does not match, and you don't configure anything. Then, the default number of connections (2) will be used which may cause requests to wait long for a connection to become available (without a proper timeout) resulting in bad responsiveness.   
```

Comment:
better? "and you don't configure anything" -> "and the intended configuration will not be used"

Reply:
check